api/document/tests/end_to_end_test.py

```diff
@@ -17,16 +18,33 @@
 EXAMPLE_DOCS_DIR = MY_DIR / '..' / '..' / 'example_docs'
 
 
-@pytest.mark.xfail(raises=AssertionError, reason="Fix this!!")
+def clean(content: bytes) -> str:
+    s = content.decode('utf-8')
+
+    # We're going to strip out `id` attributes here because our
+    # legacy XML importer seems to auto-assign ids in a different
+    # way than our serializer.
+    s = re.sub(r'id="[a-z0-9_]+', '', s)
+
+    return s
+
+
+def get_cursor_for_policy(policy: Policy) -> DocCursor:
+    docnode = DocNode.objects.filter(policy=policy, depth=0).\
+        prefetch_annotations().first()
+    cursor = DocCursor.load_from_model(docnode, subtree=False)
+    cursor.add_models(cursor.descendants().prefetch_annotations())
```

Comment:
I'm not a big fan of duplicating this code from our `document.views.TreeView.get_object()` method. Do you think we should actually have `DocNode.load_from_model` call `prefetch_annotations` on its subtree queryset automatically? Are there cases where auto-prefetching those annotations wouldn't be helpful?

Reply:
Good point; the prefetch was originally separate in case we wanted to dynamically select which annotations would be present in the result, but that's all been smashed together now. No harm making it even more direct.

Reply:
Oh I had forgotten about the dynamically selection which annotations would be present in the result!  Hmm.

Meh, I will leave it as-is for now. Maybe I will do it in a separate PR.